### PR TITLE
Add new recipe: rollbear::strong_type

### PR DESCRIPTION
**Pull request by a4z**
Specify library name and version:  **strong_type/v7**

since [strong_type](https://github.com/rollbear/strong_type) ins now available on [Compiler Explorer ](https://godbolt.org/z/6GEPz6ETe), I thought it should also be available on Conan

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/strong_type/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+required_conan_version = ">=1.50.0"
+
+
+class StrongTypeConan(ConanFile):
+    name = "strong_type"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rollbear/strong_type"
+    description = "An additive strong typedef library for C++14/17/20"
+    topics = ("cpp14", "cpp17", "strong_type")
+    license = "Boost Software License 1.0"
```
Comment (uilianries) [inline recipes/strong_type/all/conanfile.py:16]:
```suggestion
    license = "BSL-1.0"
```
https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution, only need to update the license attribute and we are good to go.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]
Thank you!

**Pull request merged**